unit/java_bytecode/java_bytecode_specialize_generics/java_bytecode_specialize_generics.cpp

```diff
@@ -0,0 +1,432 @@
+/*******************************************************************\
+
+ Module: Unit tests for specializing generic methods
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <catch.hpp>
+
+#include <istream>
+#include <memory>
+
+#include <util/config.h>
+#include <util/language.h>
+#include <util/message.h>
+#include <java_bytecode/java_bytecode_language.h>
+#include <java_bytecode/generate_java_generic_method.h>
+#include <iostream>
+#include <util/namespace.h>
+#include <util/std_types.h>
+#include <expr_iterator.h>
+
+
+SCENARIO(
+  "java_bytecode_specialize_generics",
+  "[core][java_bytecode][java_bytecode_specialize_generics]")
+{
+  std::unique_ptr<languaget>java_lang(new_java_bytecode_language());
+
+  // Configure the path loading
+  cmdlinet command_line;
+  command_line.set(
+    "java-cp-include-files",
+    "./java_bytecode/java_bytecode_specialize_generics");
+  config.java.classpath.push_back(
+    "./java_bytecode/java_bytecode_specialize_generics");
+
+  std::istringstream java_code_stream("ignored");
+  null_message_handlert message_handler;
+
+
+  // Configure the language, load the class files
+  java_lang->get_language_options(command_line);
+  java_lang->set_message_handler(message_handler);
+  java_lang->parse(java_code_stream, "generics.class");
```

Comment:
When rebased, you should be able to use:
```C++
load_java_class(
  "generics", "./java_bytecode/java_bytecode_specialize_generics");
```
To replace a bunch of these lines

Reply:
Great, I'll update with that, thanks.